CMakeLists.txt

```diff
@@ -75,6 +76,9 @@ add_subdirectory(lib/mpegts)
 
 add_subdirectory(lib/webm_parser)
 
+enable_testing()
+list(APPEND CMAKE_CTEST_ARGUMENTS "--output-on-failure")
+
```

Comment:
Please keep these within `if(NOT CMAKE_CROSSCOMPILING)` because compiling tests fails when cross-compiling:
```
if(NOT CMAKE_CROSSCOMPILING)
  enable_testing()
  list(APPEND CMAKE_CTEST_ARGUMENTS "--output-on-failure")
  add_subdirectory(src/test)
endif()
```

Reply:
Is there a way to make the tests compile while cross compiling? This would be ideal as otherwise the largest platform would not be tested.

Reply:
I guess these test are not platform-specific but it's definitely possible to fix this, will add another comment in a second.

Reply:
Great, thanks

Reply:
unless you hook up an android and iphone to jenkins, you cannot run tests for cross-compiled platforms